### PR TITLE
Filter out some warnings from tests

### PR DESCRIPTION
**Pull request by dstansby**
More stuff pulled out of #3648, this time ignoring warnings. This is stuff that I understand well and have commented the reasons for ignoring the various warnings. The one thing that isn't explicitly commented is setting up new clients in the tests, which has to be done as `hypothesis` does not automatically reset fixtures.

Again, if anything does not make sense please let me know!

### CODE REVIEW

File: sunpy/coordinates/tests/test_ephemeris.py [hunk 1]
```diff
@@ -124,6 +124,9 @@ def test_get_horizons_coord_array_time():
     assert_quantity_allclose(e[3].radius, 0.9908173*u.AU, atol=5e-7*u.AU)
 
 
+# Ignore the warning hypothesis throws, as in this case we don't care that
+# the fixutre isn't reset bewteen tests
```
Comment (nabobalis) [inline sunpy/coordinates/tests/test_ephemeris.py:128]:
```suggestion
# the fixture isn't reset between tests
```

**Comment by nabobalis** [comment]
I am not 100% sure about any cooridainte or frame warnings we skip over. @ayshih can you double check please?

**Comment by ayshih** [comment]
> @ayshih can you double check please?

Coordinate stuff looks good to me

**Comment by nabobalis** [comment]
> > @ayshih can you double check please?
> 
> Coordinate stuff looks good to me

Thanks.

**Pull request merged**